easybuild/tools/filetools.py

```diff
@@ -763,6 +763,13 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
+    # Attempt extracting the patch if it does not end in .patch (may end in .gz, .zip, etc.)
+    if not apatch.endswith(".patch"):
+        workdir = tempfile.mkdtemp(prefix='eb-patch-')
+        _log.debug("Extracting the patch to: %s", workdir)
+        # extracting the patch
+        apatch = extract_file(apatch, workdir)
```

Comment:
This should be fixed in the latest commit where I rebuild the patch name with the extract directory and the original name (minus the extension for compression)